results/16-LatentVar-metaviperCor.Rmd

```diff
@@ -0,0 +1,114 @@
+---
+title: "16-Latent Variable Metaviper Correlation"
+author: "Sara Gosline"
+date: "`r Sys.Date()`"
+output:
+  html_document:
+    code_folding: hide
+    toc: true
+    toc_float: true
+---
+
+```{r setup, include=FALSE, warning=FALSE}
+knitr::opts_chunk$set(echo = TRUE)
+##compare tumor immune scores to metaviper predictions
+
+require(synapser)
+synLogin()
+require(tidyverse)
+
+mp_scores='syn21046991'
+metaviper_scores='syn20503291'
+```
+
+### Latent Variables correlated with metaviper protein activity
+
+First we get the metaviper predictions and LV scores from Synapse
+
+```{r, warning=FALSE}
+#get immune predictions
+dtab<-synapser::synTableQuery(paste('select * from',mp_scores))$asDataFrame()%>%
+  subset(isCellLine!='TRUE')
+
+##get metaviper scores
+mtab<-read.csv(synapser::synGet(metaviper_scores)$path,sep='\t')%>%
+  rename(specimenID='sample')%>%
+  rename(protScore='counts')
+
+
+samps<-intersect(dtab$specimenID,mtab$specimenID)
+
+mp_res<-dtab%>%
+  subset(specimenID%in%samps)%>%
+  group_by(latent_var) %>%
+  mutate(sd_value = sd(value)) %>%
+  filter(sd_value > 0.05) %>%
+  ungroup()%>%
+  select(latent_var,value,tumorType,specimenID,sd_value,diagnosis)
+
+combined<-mp_res%>%inner_join(mtab,by='specimenID')
+
+```
+
+Then we can compute the correlation of each cell type measurement with each protein
+```{r, warning=FALSE}
+corVals=combined%>%group_by(latent_var,gene)%>%summarize(corVal=cor(value,protScore,use='pairwise.complete.obs'))
+
+corVals
+```
+
+### Plotting protein correlations
+We are interested in particular protein signaling activity that might give rise to specific immune phonetypes.
+
+```{r, warning=FALSE}
+##now how do we bracket them?
+##plot correlation distributions by cell type and method. 
+require(ggplot2)
```

Comment:
I can't see this plot. I think it is that the x-axis labels are too long. Perhaps something like `p<-ggplot(corVals %>% mutate(latent_var = stringr::str_trim(latent_var, 20)) + ... ` would work?

Reply:
I can't get this to work. I"m gonna keep it as is for now. We can revisit later.